### PR TITLE
Uuid [v3]

### PR DESCRIPTION
**Pull request by glima**
Changes since v2 (#656):
 - getrandom's return now checked correctly
 - char id[static <size>] used where possible
 - machine id does not deal with hyphens and case anymore
 - machine id is now a platform node
 - various other assorted fixes

### CODE REVIEW

File: src/lib/common/sol-platform-impl-linux-micro.c [hunk 1]
```diff
@@ -784,6 +784,99 @@ sol_platform_impl_set_target(const char *target)
...
+    if (!sol_util_uuid_str_valid(id))
+        return -EINVAL;
+
+    return 0;
+}
+
+int
+sol_platform_impl_get_machine_id(char id[static 33])
+{
+    const char *etc_path, *run_path;
```
Comment (lpereira) [inline src/lib/common/sol-platform-impl-linux-micro.c]: Why not `static const char *etc_path = "/foo/bar"`?

Comment (glima) [inline src/lib/common/sol-platform-impl-linux-micro.c]: Fair, I'll fix it.

**Comment by glima** [comment]
BTW, I'll leave this PR exposed a little bit more till I dive in fixes -- I'm working on something else in the meanwhile.

**Comment by glima** [comment]
All fixed, new version now.

**Pull request closed without merging**